File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    bowl = []

    for y in moves:
        y -= 1
        for x in range(len(board)):
            if board[x][y] != 0:
                if len(bowl) > 0 and bowl[-1] == board[x][y]:
                    bowl.pop()
                    answer += 2

                else:
                    bowl.append(board[x][y])

                board[x][y] = 0
                break

    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    vector<int> stack;
    
    for(int i=0; i<moves.size(); i++){
        int y = moves[i]-1;
        
        for(int x=0; x<board.size(); x++){
            if(board[x][y] != 0){
                int pick_up = board[x][y];
                if ((!stack.empty()) && (pick_up == stack.back())){
                    stack.pop_back();
                    answer+=2;
                }
                else{
                    stack.push_back(pick_up);
                }
                board[x][y] = 0;
                break;
            }
        }
    }
    
    return answer;
}